packages/p2p/src/components/buy-sell/buy-sell-form.jsx

```diff
@@ -222,43 +222,54 @@ const BuySellForm = props => {
-                                                return matching_payment_methods.length > 0 ? (
-                                                    matching_payment_methods.map(payment_method => (
-                                                        <PaymentMethodCard
-                                                            is_vertical_ellipsis_visible={false}
-                                                            key={key}
-                                                            medium
-                                                            onClick={() => onClickPaymentMethodCard(payment_method)}
-                                                            payment_method={payment_method}
-                                                            style={
-                                                                selected_methods.includes(payment_method.ID)
-                                                                    ? style
-                                                                    : {}
-                                                            }
-                                                        />
-                                                    ))
-                                                ) : (
-                                                    <PaymentMethodCard
-                                                        add_payment_method={add_payment_method}
-                                                        is_add={true}
-                                                        key={key}
-                                                        medium
-                                                        onClickAdd={() => {
-                                                            my_profile_store.setSelectedPaymentMethodDisplayName(
-                                                                add_payment_method
+                                        <Observer>
+                                            {() => (
+                                                <div className='buy-sell__modal--sell-payment-methods'>
+                                                    {payment_method_names?.map((add_payment_method, key) => {
+                                                        const {
+                                                            advertiser_payment_methods_list,
+                                                            setSelectedPaymentMethodDisplayName,
+                                                            setShouldShowAddPaymentMethodForm,
+                                                        } = my_profile_store;
+                                                        const matching_payment_methods =
+                                                            advertiser_payment_methods_list.filter(
+                                                                pm => pm.display_name === add_payment_method
```

Comment:
maybe instead of saying pm, it is best to name it in full 'payment_method' as it would be easier to understand for those who are new to the code